lib/matplotlib/widgets.py

```diff
@@ -1019,22 +1046,33 @@ def __init__(self, ax, labels, actives=None, *, useblit=True):
         self._background = None
 
         ys = np.linspace(1, 0, len(labels)+2)[1:-1]
-        text_size = mpl.rcParams["font.size"] / 2
 
+        label_props = _expand_text_props(label_props)
         self.labels = [
             ax.text(0.25, y, label, transform=ax.transAxes,
-                    horizontalalignment="left", verticalalignment="center")
-            for y, label in zip(ys, labels)]
-
-        self._squares = ax.scatter(
-            [0.15] * len(ys), ys, marker='s', s=text_size**2,
-            c="none", linewidth=1, transform=ax.transAxes, edgecolor="k"
-        )
-        self._crosses = ax.scatter(
-            [0.15] * len(ys), ys, marker='x', linewidth=1, s=text_size**2,
-            c=["k" if active else "none" for active in actives],
-            transform=ax.transAxes, animated=self._useblit,
-        )
+                    horizontalalignment="left", verticalalignment="center",
+                    **props)
+            for y, label, props in zip(ys, labels, label_props)]
+        text_size = np.array([text.get_fontsize() for text in self.labels]) / 2
+
+        frame_props = {
+            'marker': 's', 's': text_size**2,
+            'facecolor': 'none', 'linewidth': 1, 'edgecolor': 'k',
```

Comment:
I think this would be more readable with each dict entry on a new line - that's an optional suggestion though 😃